music21/expressions.py

```diff
@@ -325,6 +325,119 @@ def nextMark(self):
         return RehearsalMark(self.nextContent(), numbering=self.numbering)
 
 
+# ------------------------------------------------------------------------------
+class PedalType(common.StrEnum):
+    Sustain = 'sustain'
+    Sostenuto = 'sostenuto'
+    Soft = 'soft'
+    Silent = 'silent'
+
+class PedalForm(common.StrEnum):
+    Line = 'line'
+    Symbol = 'symbol'
+    SymbolAlt = 'symbolalt'
+    SymbolLine = 'symbolline'
+
+class PedalMark(spanner.Spanner):
+    '''
+        A pedal mark spanner contains a pedaled set of notes.
+        The spanner starts at the moment of down-pedal, and
+        ends at the moment of up-pedal.  There can be any
+        number of pedal "bounces" in the middle of the spanner.
+
+        The visual form of the pedal mark can vary.  The following
+        examples use a pedal mark with one "bounce" in the middle::
+
+            Pedal marks can be lines:            |_______^________|
+            Pedal marks can be normal symbolic:  Ped.    * Ped.   *
+            Pedal marks can be altered symbolic: Ped.    Ped.     *
+            Pedal marks can be symbol and line:  Ped.____^________|
+
+        Pedal marks, whether lines, symbols, or a combination, can
+        represent the sustain pedal (Ped.), the sostenuto pedal
+        (Sost.), the soft (una corda) pedal, or (more rarely) the
+        silent (muting) pedal.
+
+        Pedal marks that are symbolic can be abbreviated: e.g. P. instead
+        of Ped., S. instead of Sost.
+
+        Pedal marks that are lines can have non-printed portions
+        (gaps) in them; these are usually started with "simile",
+        but not necessarily.
+    '''
+    def __init__(
+        self,
+        *spannedElements,
+        **keywords
+    ) -> None:
+        super().__init__(*spannedElements, **keywords)
+        from music21 import note
+        self.fillElementTypes = [note.GeneralNote]
+
+        self._pedalType: PedalType|None = None
+        self._pedalForm: PedalForm|None = None
+        self.abbreviated: bool = False
+        self.placement: str|None = None
+        self.startOffset: OffsetQL|None = None  # only needed temporarily by xmlToM21.py
+
+    @property
+    def pedalType(self) -> PedalType|None:
+        return self._pedalType
+
+    @pedalType.setter
+    def pedalType(self, newType: PedalType|None):
+        if newType is not None and newType not in PedalType:
+            raise ValueError(f'invalid pedalType {newType!r}')
+        self._pedalType = newType
```

Comment:
Properties are fine here, but just wanted to note that with modern Python typing, we can leave type error detecting to the static checker and not use getters and setters for mere type validation.

Reply:
Removed them.